components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/dao/OAuthUserConsentedScopesDAOImpl.java

```diff
@@ -0,0 +1,359 @@
+                        boolean consent = rs.getBoolean(2);
+                        if (consent) {
+                            approvedScopes.add(scope);
+                        } else {
+                            disapprovedScopes.add(scope);
+                        }
+                    }
+                }
+            }
+            userScopeConsent.setApprovedScopes(approvedScopes);
+            userScopeConsent.setDeniedScopes(disapprovedScopes);
+            return userScopeConsent;
+        } catch (SQLException e) {
+            String msg = "Error occurred while retrieving scope consents for  userId  :" + userId + " and appId: " +
+                    appId + " and " + "tenantId : " + tenantId;
+            throw new IdentityOAuth2ScopeConsentException(msg, e);
+        }
+    }
+
+    @Override
+    public List<UserApplicationScopeConsentDO> getUserConsents(String userId, int tenantId)
+            throws IdentityOAuth2ScopeConsentException {
+
+        if (log.isDebugEnabled()) {
+            log.debug("Get user consented scopes for user with userId  :" + userId + " in tenantId : " + tenantId);
+        }
+        Map<String, UserApplicationScopeConsentDO> userScopeConsentsMap = new HashMap<>();
+        try (Connection conn = IdentityDatabaseUtil.getDBConnection(false)) {
+            try (PreparedStatement ps = conn.prepareStatement(SQLQueries.GET_OAUTH2_USER_CONSENTS)) {
+                ps.setString(1, userId);
+                ps.setInt(2, tenantId);
+                try (ResultSet rs = ps.executeQuery()) {
+                    while (rs.next()) {
+                        String appId = rs.getString(1);
+                        String scope = rs.getString(2);
+                        boolean consent = rs.getBoolean(3);
+                        userScopeConsentsMap.putIfAbsent(appId, new UserApplicationScopeConsentDO(appId));
+                        if (consent) {
+                            userScopeConsentsMap.get(appId).getApprovedScopes().add(scope);
+                        } else {
+                            userScopeConsentsMap.get(appId).getDeniedScopes().add(scope);
+                        }
+                    }
+                }
+            }
+            return new ArrayList<>(userScopeConsentsMap.values());
+        } catch (SQLException e) {
+            String msg = "Error occurred while retrieving scope consents for  userId  :" + userId + " in tenantId : "
+                    + tenantId;
+            throw new IdentityOAuth2ScopeConsentException(msg, e);
+        }
+    }
+
+    @Override
+    public void addUserConsentForApplication(String userId, int tenantId,
+                                             UserApplicationScopeConsentDO userConsent)
+            throws IdentityOAuth2ScopeConsentException {
+
+        if (log.isDebugEnabled()) {
+            log.debug("Adding scope consents for userId  :" + userId + " and appId: " + userConsent.getAppId() +
```

Comment:
formatting

Reply:
Fixed in https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/1548/commits/6dbf4cf4e588670e742bf004582eb75fae29566e